src/isar/state_machine/state_machine.py

```diff
@@ -301,7 +307,7 @@ def _stop(self) -> None:
 
     def _initiate_step_failed(self) -> None:
         self.current_step.status = StepStatus.Failed
-        self.current_mission.status = MissionStatus.Failed
+        self.current_mission.status = MissionStatus.PartiallySuccessful
```

Comment:
Should we set the missions status here or only in the `_mission_finished()` method?
Maybe it made sense when it failed because it would be aborted, but setting it to partially successful implies it will continue, so maybe setting the final status doesn't make sense anymore.

Reply:
Will fix